torch/csrc/jit/tensorexpr/llvm_codegen.cpp

```diff
@@ -818,7 +818,7 @@ void LLVMCodeGenImpl::visit(const Load* v) {
   if (v->dtype().lanes() == 1) {
     v->base_handle()->accept(this);
     auto base = this->value_;
-    v->index()->accept(this);
+    v->flattened_index()->accept(this);
```

Comment:
I think the index-flattening should be part of the common-mode lowering, before sending to individual backends. So the backends do not have to deal with this complexity. In other words, 

* Schedule transformation: 
  * All the indices are multi-dimensional tensor index
* When lower is called: 
  * First, common-mode lower
    * Convert the tensor indices into flattened index
    * Apply transformations such as vectorization
  * Send to device-specific codegen, where the index is flattened, and vectorized.

Reply:
It works now approximately as you described, this call here does not perform the flattening - what it does is it tries to access the flattened index. If the indexes have not been lowered at that point, this call will error out.